js/memory.js

```diff
@@ -0,0 +1,76 @@
+// function createGame(){
+// 	new squareFactory()
+// }
+var squareFactory;
+(squareFactory = function squareFactory(){
+	var squareObjects = []
+	this.squareElements = $('.square');
+	this.squareObjects = squareObjects;
+	this.initSquareObjects = function initSquareObjects() {
+		for (var idx in this.squareElements) {
```

Comment:
I'd recommend using a normal `for` loop for arrays.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in#Array_iteration_and_for...in